server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -356,7 +371,20 @@ public void onResponse(DiscoveryNode remoteNode) {
 
                 @Override
                 public void onFailure(Exception e) {
-                    logger.debug(() -> new ParameterizedMessage("{} connection failed", Peer.this), e);
+                    if (verboseFailureLogging) {
```

Comment:
In order to still be able to get stack traces out, we should either do following:
```suggestion
                    if (verboseFailureLogging && logger.isDebugEnabled() == false) {
```

or let the `logger.debug` log regardless of whether we enabled verbose logging.

Reply:
Ah that's a good point. I'd rather not have the two messages, but I think we should still log at `WARN` even if the level is `DEBUG`. See 3f032848b2b9d0ee778abe347e944f221d013680.